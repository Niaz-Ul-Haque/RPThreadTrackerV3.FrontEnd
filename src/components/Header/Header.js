import React, { Component } from 'react';
import {
	Nav,
	NavItem,
	NavLink,
	Badge,
	NavbarToggler,
	NavbarBrand,
	Dropdown,
	DropdownToggle,
	DropdownMenu,
	DropdownItem
} from 'reactstrap';

class Header extends Component {
	constructor(props) {
		super(props);

		this.toggle = this.toggle.bind(this);
		this.messagesToggle = this.messagesToggle.bind(this);
		this.state = {
			dropdownOpen: false,
			messagesDropdownOpen: false
		};
	}

	toggle() {
		this.setState({
			dropdownOpen: !this.state.dropdownOpen
		});
	}

	sidebarToggle(e) {
		e.preventDefault();
		document.body.classList.toggle('sidebar-hidden');
	}

	messagesToggle() {
		this.setState({
			messagesDropdownOpen: !this.state.messagesDropdownOpen
		});
	}

	sidebarMinimize(e) {
		e.preventDefault();
		document.body.classList.toggle('sidebar-minimized');
	}

	mobileSidebarToggle(e) {
		e.preventDefault();
		document.body.classList.toggle('sidebar-mobile-show');
	}

	asideToggle(e) {
		e.preventDefault();
		document.body.classList.toggle('aside-menu-hidden');
	}

	render() {
		return (
			<header className="app-header navbar">
				<NavbarToggler className="d-lg-none" onClick={this.mobileSidebarToggle}>&#9776;</NavbarToggler>
				<NavbarBrand href="#"></NavbarBrand>
				<NavbarToggler className="d-md-down-none mr-auto" onClick={this.sidebarToggle}>&#9776;</NavbarToggler>
				<Nav className="ml-auto" navbar>
					<NavItem className="d-md-down-none">
						<Dropdown isOpen={this.state.messagesDropdownOpen} toggle={this.messagesToggle}>
							<DropdownToggle className="nav-link">
								<NavLink href="#"><i className="icon-bell"></i><Badge pill color="danger">5</Badge></NavLink>
							</DropdownToggle>
							<DropdownMenu right className={this.state.messagesDropdownOpen ? 'show' : ''}>
								<DropdownItem><i className="fa fa-lock"></i> Logout</DropdownItem>
							</DropdownMenu>
						</Dropdown>
					</NavItem>
					<NavItem>
						<Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
							<DropdownToggle className="nav-link dropdown-toggle">
								<span className="d-md-down-none">blackjackkent</span>
							</DropdownToggle>
							<DropdownMenu right className={this.state.dropdownOpen ? 'show' : ''}>
								<DropdownItem><i className="fa fa-lock"></i> Logout</DropdownItem>
							</DropdownMenu>
						</Dropdown>
					</NavItem>
				</Nav >
			</header >
		)
	}
}

export default Header;
