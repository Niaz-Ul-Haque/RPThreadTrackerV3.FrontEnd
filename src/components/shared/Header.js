import React, { Component } from 'react';
import {
	Nav,
	NavItem,
	NavLink,
	Badge,
	NavbarToggler,
	NavbarBrand,
	Dropdown,
	DropdownToggle,
	DropdownMenu,
	DropdownItem
} from 'reactstrap';

class Header extends Component {
	constructor(props) {
		super(props);

		this.toggle = this.toggle.bind(this);
		this.messagesToggle = this.messagesToggle.bind(this);
		this.state = {
			dropdownOpen: false,
			messagesDropdownOpen: false
		};
	}

	sidebarToggle(e) {
		e.preventDefault();
		document.body.classList.toggle('sidebar-hidden');
	}

	asideToggle(e) {
		e.preventDefault();
		document.body.classList.toggle('aside-menu-hidden');
	}

	messagesToggle() {
		this.setState({
			messagesDropdownOpen: !this.state.messagesDropdownOpen
		});
	}

	sidebarMinimize(e) {
		e.preventDefault();
		document.body.classList.toggle('sidebar-minimized');
	}

	mobileSidebarToggle(e) {
		e.preventDefault();
		document.body.classList.toggle('sidebar-mobile-show');
	}

	toggle() {
		this.setState({
			dropdownOpen: !this.state.dropdownOpen
		});
	}

	render() {
		return (
			<header className="app-header navbar">
				<NavbarToggler className="d-lg-none" onClick={this.mobileSidebarToggle}>
					&#9776;
				</NavbarToggler>
				<NavbarBrand href="#"></NavbarBrand>
				<NavbarToggler className="d-md-down-none mr-auto" onClick={this.sidebarToggle}>
					&#9776;
				</NavbarToggler>
				<Nav className="ml-auto" navbar>
					<NavItem className="d-md-down-none">
						<NavLink href="#" onClick={this.asideToggle}>
							<i className="icon-bell"></i>
							<Badge pill color="danger">5</Badge>
						</NavLink>
					</NavItem>
					<NavItem>
						<Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
							<DropdownToggle className="nav-link dropdown-toggle">
								<span className="d-md-down-none">blackjackkent</span>
							</DropdownToggle>
							<DropdownMenu right className={this.state.dropdownOpen ? 'show' : ''}>
								<DropdownItem><i className="fa fa-lock"></i> Logout</DropdownItem>
							</DropdownMenu>
						</Dropdown>
					</NavItem>
				</Nav >
			</header >
		)
	}
}

export default Header;
