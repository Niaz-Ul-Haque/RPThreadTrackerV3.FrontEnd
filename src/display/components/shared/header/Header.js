import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
	Nav,
	NavItem,
	NavLink,
	Badge,
	NavbarToggler,
	NavbarBrand,
	Dropdown,
	DropdownToggle,
	DropdownMenu,
	DropdownItem
} from 'reactstrap';
import { connect } from 'react-redux';
import { toggleSidebar, toggleNewsAside, toggleHeaderDropdown, toggleMobileSidebar } from '../../../../state/ui/actions';

const propTypes = {
	// eslint-disable-next-line react/no-unused-prop-types
	isNewsAsideOpen: PropTypes.bool.isRequired,
	// eslint-disable-next-line react/no-unused-prop-types
	isSidebarOpen: PropTypes.bool.isRequired,
	// eslint-disable-next-line react/no-unused-prop-types
	isMobileSidebarOpen: PropTypes.bool.isRequired,
	isHeaderDropdownOpen: PropTypes.bool.isRequired,
	dispatch: PropTypes.func.isRequired
};

function mapStateToProps(state) {
	const { ui } = state;
	const {
		isNewsAsideOpen, isSidebarOpen, isHeaderDropdownOpen, isMobileSidebarOpen
	} = ui;
	return {
		isNewsAsideOpen, isSidebarOpen, isHeaderDropdownOpen, isMobileSidebarOpen
	};
}

class Header extends Component {
	constructor(props) {
		super(props);
		this.asideToggle = this.asideToggle.bind(this);
		this.sidebarToggle = this.sidebarToggle.bind(this);
		this.mobileSidebarToggle = this.mobileSidebarToggle.bind(this);
		this.headerDropdownToggle = this.headerDropdownToggle.bind(this);
	}

	componentDidMount() {
		this.loadBodyClasses(this.props);
	}

	componentWillReceiveProps(nextProps) {
		this.loadBodyClasses(nextProps);
	}

	loadBodyClasses(props) {
		document.body.classList.toggle('sidebar-hidden', !props.isSidebarOpen);
		document.body.classList.toggle('aside-menu-hidden', !props.isNewsAsideOpen);
		document.body.classList.toggle('sidebar-mobile-show', props.isMobileSidebarOpen);
	}

	sidebarToggle() {
		const { dispatch } = this.props;
		dispatch(toggleSidebar());
	}

	asideToggle() {
		const { dispatch } = this.props;
		dispatch(toggleNewsAside());
	}

	mobileSidebarToggle() {
		const { dispatch } = this.props;
		dispatch(toggleMobileSidebar());
	}

	headerDropdownToggle() {
		const { dispatch } = this.props;
		dispatch(toggleHeaderDropdown());
	}

	render() {
		return (
			<header className="app-header navbar">
				<NavbarToggler className="d-lg-none" onClick={this.mobileSidebarToggle}>
					&#9776;
				</NavbarToggler>
				<NavbarBrand href="#" />
				<NavbarToggler className="d-md-down-none mr-auto" onClick={this.sidebarToggle}>
					&#9776;
				</NavbarToggler>
				<Nav className="ml-auto" navbar>
					<NavItem>
						<NavLink href="#" onClick={this.asideToggle}>
							<i className="icon-bell" />
							<Badge pill color="danger">5</Badge>
						</NavLink>
					</NavItem>
					<NavItem>
						<Dropdown isOpen={this.props.isHeaderDropdownOpen} toggle={this.headerDropdownToggle}>
							<DropdownToggle className="nav-link dropdown-toggle">
								<span className="d-md-down-none">blackjackkent</span>
							</DropdownToggle>
							<DropdownMenu right className={this.props.isHeaderDropdownOpen ? 'show' : ''}>
								<DropdownItem><i className="fa fa-lock" /> Logout</DropdownItem>
							</DropdownMenu>
						</Dropdown>
					</NavItem>
				</Nav>
			</header>
		);
	}
}

Header.propTypes = propTypes;

export default connect(mapStateToProps)(Header);
