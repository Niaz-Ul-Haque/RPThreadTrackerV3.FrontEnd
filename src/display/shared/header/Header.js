import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
	Nav,
	NavItem,
	NavLink,
	Badge,
	Button,
	NavbarToggler,
	Dropdown,
	DropdownToggle,
	DropdownMenu,
	DropdownItem
} from 'reactstrap';
import LoadingIndicator from '../LoadingIndicator';

const propTypes = {
	asideToggle: PropTypes.func.isRequired,
	headerDropdownToggle: PropTypes.func.isRequired,
	isHeaderDropdownOpen: PropTypes.bool.isRequired,
	logout: PropTypes.func.isRequired,
	mobileSidebarToggle: PropTypes.func.isRequired,
	newsUnreadCount: PropTypes.number.isRequired,
	openEditCharacterModal: PropTypes.func.isRequired,
	openNewThreadModal: PropTypes.func.isRequired,
	sidebarToggle: PropTypes.func.isRequired,
	threadsLoading: PropTypes.bool.isRequired,
	user: PropTypes.shape({
		id: PropTypes.string
	}).isRequired
};

const Header = (props) => {
	const {
		asideToggle,
		headerDropdownToggle,
		isHeaderDropdownOpen,
		logout,
		mobileSidebarToggle,
		newsUnreadCount,
		openEditCharacterModal,
		openNewThreadModal,
		sidebarToggle,
		threadsLoading,
		user
	} = props;

	return (
		<header className="app-header navbar">
			<NavbarToggler className="d-lg-none" onClick={mobileSidebarToggle}>
				&#9776;
			</NavbarToggler>
			<Link href="/" className="navbar-brand" to="/">RPTHREADTRACKER</Link>
			<NavbarToggler className="d-md-down-none" onClick={sidebarToggle}>
				&#9776;
			</NavbarToggler>
			<Nav className="d-md-down-none ml-4" navbar>
				<NavItem>
					<Button color="primary" onClick={() => openEditCharacterModal({})}>Add Character</Button>
				</NavItem>
				<NavItem>
					<Button className="ml-4" color="primary" onClick={() => openNewThreadModal({})}> Track New Thread</Button>
				</NavItem>
				{threadsLoading && <LoadingIndicator className="invert" />}
			</Nav>
			<Nav className="ml-auto" navbar>
				<NavItem>
					<NavLink href="#" onClick={asideToggle}>
						<i className="icon-bell" />
						{newsUnreadCount > 0 &&
							<Badge pill color="danger">{newsUnreadCount}</Badge>
						}
					</NavLink>
				</NavItem>
				<NavItem>
					<Dropdown isOpen={isHeaderDropdownOpen} toggle={headerDropdownToggle}>
						<DropdownToggle className="nav-link dropdown-toggle">
							<span className="d-md-down-none">{user ? user.userName : ''}</span>
						</DropdownToggle>
						<DropdownMenu right className={isHeaderDropdownOpen ? 'show' : ''}>
							<DropdownItem className="hidden-lg-up">
								Track New Thread
							</DropdownItem>
							<DropdownItem className="hidden-lg-up">
								Add Character
							</DropdownItem>
							<DropdownItem onClick={logout}>
								Logout
							</DropdownItem>
						</DropdownMenu>
					</Dropdown>
				</NavItem>
			</Nav>
		</header>
	);
};

Header.propTypes = propTypes;

export default Header;
