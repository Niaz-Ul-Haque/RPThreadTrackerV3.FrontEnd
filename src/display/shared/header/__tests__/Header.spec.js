// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';
// #endregion imports

// #region mocks
jest.mock('../components', () => ({
	HeaderLogoBlock: () => 'HeaderLogoBlock',
	HeaderAsideToggle: () => 'HeaderAsideToggle',
	HeaderProfileDropdown: () => 'HeaderProfileDropdown',
	HeaderAddMenuDropdown: () => 'HeaderAddMenuDropdown'
}));
// #endregion mocks

const createTestProps = (propOverrides) => ({
	asideToggle: jest.fn(),
	headerProfileDropdownToggle: jest.fn(),
	headerAddMenuDropdownToggle: jest.fn(),
	isHeaderProfileDropdownOpen: true,
	isHeaderAddMenuDropdownOpen: true,
	logout: jest.fn(),
	mobileSidebarToggle: jest.fn(),
	newsUnreadCount: 2,
	openUpsertCharacterModal: jest.fn(),
	openNewThreadModal: jest.fn(),
	navigateToSettings: jest.fn(),
	navigateToTools: jest.fn(),
	navigateToHelp: jest.fn(),
	sidebarToggle: jest.fn(),
	isSidebarOpen: true,
	isMobileSidebarOpen: true,
	isNewsAsideOpen: true,
	user: {
		id: '12345'
	},
	...propOverrides
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot', () => {
			const props = createTestProps();
			const jsx = <Header {...props} />;
			const element = shallow(jsx);
			expect(element).toMatchSnapshot();
		});
	});
});
