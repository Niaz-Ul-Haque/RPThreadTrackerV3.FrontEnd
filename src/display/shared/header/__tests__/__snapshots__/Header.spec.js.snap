// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<Style
  className="app-header navbar"
>
  <HeaderLogoBlock
    mobileSidebarToggle={[MockFunction]}
    sidebarToggle={[MockFunction]}
  />
  <Nav
    className="ml-auto"
    navbar={true}
    tag="ul"
  >
    <HeaderAsideToggle
      asideToggle={[MockFunction]}
      newsUnreadCount={2}
    />
    <HeaderAddMenuDropdown
      headerAddMenuDropdownToggle={[MockFunction]}
      isHeaderAddMenuDropdownOpen={true}
      logout={[MockFunction]}
      openNewThreadModal={[MockFunction]}
      openUpsertCharacterModal={[MockFunction]}
      user={
        Object {
          "id": "12345",
        }
      }
    />
    <HeaderProfileDropdown
      headerProfileDropdownToggle={[MockFunction]}
      isHeaderProfileDropdownOpen={true}
      logout={[MockFunction]}
      user={
        Object {
          "id": "12345",
        }
      }
    />
  </Nav>
</Style>
`;
